### DISCUSSION TITLE
phonoLammps+deepMD-kit,error

### DISCUSSION THREAD

**Discussion by langzii** [deepmodeling/deepmd-kit#2022] [Q&A] [answered]
Run this command " phonolammps in.lammps -c POSCAR --dim 5 5 5" can get force constants using phonoLammps with pair_style in lammps such as tersoff , but when I use deepMD poteneial energy can't get force constants file.

**Comment by langzii** [comment, marked as answer]
The Lammps inpute file are as following,
units		metal
boundary        p p p


atom_style	atomic

read_data      test.data

pair_style	deepmd ./graph.pb
pair_coeff      * *

neighbor        2.0 bin

**Reply by njzjz** [reply]
Can you take a look at `log.lammps`?

When LAMMPS is not built with `-D LAMMPS_EXCEPTIONS=yes`, its Python library will not throw errors on the screen.

**Reply by langzii** [reply]
My log.lammps are as following,
LAMMPS (29 Sep 2021)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
plugin load libdeepmd_lmp.so
Loading plugin: deepmd pair style v2.0 by Han Wang
Loading plugin: compute deeptensor/atom v2.0 by Han Wang
Loading plugin: fix dplr v2.0 by Han Wang
Total wall time: 0:00:00


since phonolammps can't use DeepMD-potential ,I don't have a method to use my  machine learning potential energy.

**Reply by njzjz** [reply]
You can add `--logshow` for detailed log. See https://phonolammps.readthedocs.io/en/master/troubleshooting.html

**Reply by langzii** [reply]
Add this tag my error is
ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:278)
Last command: pair_style	deepmd ./graph.pb

**But** use lammps can read this deepmd potential. The result is no output file i.e. FORCE_CONSTANTS

**Reply by langzii** [reply]
LAMMPS (29 Sep 2021)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Ignoring load of pair style deepmd: must unload existing deepmd plugin first
Ignoring load of compute style deepmd: must unload existing deepmd plugin first
Ignoring load of fix style deepmd: must unload existing deepmd plugin first
Reading data file ...
  triclinic box = (0.0000000 0.0000000 0.0000000) to (5.4335936 5.4335936 2.5713504) with tilt (0.0000000 0.0000000 0.0000000)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  12 atoms
  read_data CPU = 0.000 seconds
ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:278)
Last command: pair_style	deepmd ./graph.pb

**Reply by njzjz** [reply]
The plugin may not be loaded correctly.
You can add `clear` and `plugin load libdeepmd_lmp.so` to the beginning of your input file.

**Reply by langzii** [reply]
I took the advice ,but the problem are become
LAMMPS (29 Sep 2021)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Ignoring load of pair style deepmd: must unload existing deepmd plugin first
Ignoring load of compute style deepmd: must unload existing deepmd plugin first
Ignoring load of fix style deepmd: must unload existing deepmd plugin first
**Ignoring load of pair style deepmd: must unload existing deepmd plugin first
Ignoring load of compute style deepmd: must unload existing deepmd plugin first**
Ignoring load of fix style deepmd: must unload existing deepmd plugin first
Reading data file ...
  triclinic box = (0.0000000 0.0000000 0.0000000) to (5.4335936 5.4335936 2.5713504) with tilt (0.0000000 0.0000000 0.0000000)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  12 atoms
  read_data CPU = 0.000 seconds
ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:278)
Last command: pair_style	deepmd ./graph.pb

and the message tell unload deepmd, I don't know how to fix the problem.

**Reply by njzjz** [reply]
It looks like the bug reported in https://github.com/lammps/lammps/issues/3000. You may need to use newer lammps.

**Reply by langzii** [reply]
I use the new Lammps but error hapen in deepMD-kit
AttributeError: /home/wd/deepmd-kit/lib/python3.9/lib-dynload/../../liblammps.so: undefined symbol: lammps_flush_buffers
what wrong about this error?

**Reply by langzii** [reply]
Python 3.9.7 (default, Sep 16 2021, 13:09:58) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from lammps import lammps
>>> lmp=lammps()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/wd/Desktop/LFD/soft/lammps/lammps-15Sep2022/python/lammps/core.py", line 167, in __init__
    self.lib.lammps_flush_buffers.argtypes = [c_void_p]
  File "/home/wd/deepmd-kit/lib/python3.9/ctypes/__init__.py", line 395, in __getattr__
    func = self.__getitem__(name)
  File "/home/wd/deepmd-kit/lib/python3.9/ctypes/__init__.py", line 400, in __getitem__
    func = self._FuncPtr((name_or_ordinal, self))
**AttributeError: /home/wd/deepmd-kit/lib/python3.9/lib-dynload/../../liblammps.so: undefined symbol: lammps_flush_buffers**
>>> exit()

**Reply by langzii** [reply]
Thanks for your help, I have solved the problem! I don't add **clear** in the inpute file.

**Comment by langzii** [comment]
Thanks your attention, I open the log.lammps file and it just repeats what's on the screen








------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "deepmodeling/deepmd-kit"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年10月25日(星期二) 中午11:13
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [deepmodeling/deepmd-kit] phonoLammps+deepMD-kit,error (Discussion #2022)





 
Can you take a look at log.lammps?
 
When LAMMPS is not built with -D LAMMPS_EXCEPTIONS=yes, its Python library will not throw errors on the screen.
